### PR TITLE
Update Sentry to latest version

### PR DESCRIPTION
**Pull request by kathaypacific**
### Description

This PR updates Sentry to the latest version. According to the migration docs, there were no changes needed from our side (there were very few breaking changes and we are not using the changed features). 

https://docs.sentry.io/platforms/react-native/migration/

### Tested

Rebuilt both iOS and Android apps, manually (created and) triggered an error and verified that it came through in Sentry.

### How others should test

There should be no change to the app behaviour with this PR, so verify that Sentry is working as expected. Errors that you trigger should be received by Sentry as usual.

### Related issues

Prepares for #1461

### CODE REVIEW

File: packages/mobile/ios/Podfile.lock [hunk 1]
```diff
@@ -1101,19 +1101,19 @@ SPEC CHECKSUMS:
...
-  Sentry: 388c9dc093b2fd3a264466a5c5b21e25959610a9
+  Sentry: 5c5dd4005f3b7b9765d5a8871232cddbd0d888b7
   Toast: 91b396c56ee72a5790816f40d3a94dd357abc196
   TOCropViewController: 3105367e808b7d3d886a74ff59bf4804e7d3ab38
   Yoga: 4bd86afe9883422a7c4028c00e34790f560923d6
   YogaKit: f782866e155069a2cca2517aafea43200b01fd5a
 
 PODFILE CHECKSUM: d88bd91187a580924250d0d0c8281a6e164b4a68
 
-COCOAPODS: 1.10.1
```
Comment (jeanregisser) [inline packages/mobile/ios/Podfile.lock:1119]: We should be using our "pinned" version of CocoaPods from our Gemfile.lock.
Looks like `pod ...` was run instead of `bundle exec pod ...`

It's fine to update CocoaPods, but would be preferable in a different PR.

**Review by jeanregisser — COMMENTED** [review]
Great! 👍 
Almost good, just a small thing about the CocoaPods version.

File: packages/mobile/ios/Podfile.lock (hunk 1 above)
Comment (kathaypacific) [reply packages/mobile/ios/Podfile.lock]: good catch! that's exactly what happened indeed (had some issues installing pods on my machine, i did end up doing a `pod install` instead of `bundle exec pod install` - i got lost in the weeds) 

for this i uninstalled cocoapods + reinstalled to the version we have in the podfile.lock + ran `pod update Sentry`. ran `bundle exec pod install` afterwards to check no more changes were added.

**Review by jeanregisser — APPROVED** [review]

**Comment by ValoraQA** [comment]
Hey @MuckT , Please let us know do we need to perform testing from our side.

**Comment by kathaypacific** [comment]
hey @ValoraQA no testing is needed from your side for this one :)
